packages/components/inputs/radio-input/src/radio-option.styles.ts

```diff
@@ -133,25 +120,96 @@ const getLabelCursor = (props: TStylesProps) => {
   return 'pointer';
 };
 
-const getLabelStyles = (props: TStylesProps) => css`
-  align-items: center;
-  color: ${getLabelColor(props)};
-  cursor: ${getLabelCursor(props)};
+const RadioOptionKnob = styled.div<TStylesProps>`
+  width: ${designTokens.heightForRadioInputOptionWhenChecked};
+  height: ${designTokens.heightForRadioInputOptionWhenChecked};
+  border-radius: 50%;
+  background-color: ${(props) => getKnobColor(props)};
+`;
+
+const RadioOptionBorder = styled.div<TStylesProps>`
+  width: ${designTokens.heightForRadioInputOption};
+  height: ${designTokens.heightForRadioInputOption};
+  border-radius: 50%;
+  background-color: ${designTokens.backgroundColorForInput};
+  border: ${designTokens.borderForRadioInputOption} solid
+    ${(props) => getBorderColor(props)};
   display: flex;
-  position: relative;
-  &:hover svg *[data-style='radio-option__border'] {
-    stroke: ${getSvgLabelBorderStroke(props)};
+  align-items: center;
+  justify-content: center;
+`;
+
+const Input = styled.input`
+  &:focus + div > ${RadioOptionBorder} {
+    border-color: ${designTokens.borderColorForInputWhenFocused};
+  }
+`;
+
+const getNewThemeHoverStyles = (props: TStylesProps) => {
+  const hoverStyles = css`
+    [class*='getContainerStyles'] {
+      border-color: ${designTokens.colorNeutral90};
+    }
+  `;
+
+  return [
+    !props.isDisabled &&
+      !props.isReadOnly &&
+      /* prettier-ignore */
+      css`
+        &:hover ${hoverStyles};
+      `,
+    props.isHovered && hoverStyles,
+  ];
+};
+
+const getDefaultThemeHoverAndFocusStyles = (props: TStylesProps) => css`
+  &:hover ${RadioOptionBorder} {
+    border-color: ${getLabelBorderColor(props)};
   }
```

Comment:
Also, this is not directly mentioned in the designs, but for the there is a `focus` state indication (when the input is additionally in the `readonly`/`error`/`warning` state):
<img width="134" alt="Screenshot 2023-02-21 at 15 26 22" src="https://user-images.githubusercontent.com/49066275/220371938-851bd38c-777d-43ec-ba3c-3a894c80c160.png">

Otherwise, there is no visual information for the user that the input has focus.
Perhaps this is not expected.
cc @FilPob

Reply:
@kark The indication of a focus state is given by the green border around the label now right? I think given that, we don't need a `focused` indication in the radio input itself

Reply:
Right 👍 done here 48efc33eb644a73ce2f3474ded55977d4c51c90c